kore/src/Kore/Step/Function/Evaluator.hs

```diff
@@ -121,6 +127,8 @@ evaluateApplication
           -- have a separate evaluator for startup.
           , (not . null) axiomIdToEvaluator
           = criticalMissingHook symbol hook
+          | (not . null) axiomIdToEvaluator
+          = warnUnhookedSymbol symbol $> unevaluated
```

Comment:
This isn't going to do what we want. This simplifier is invoked whenever the function is left unevaluated. For a hook, that means the hook is missing, but for a user-defined symbol there could be many reasons we can't evaluate it yet. I think we should move both of these checks [here](https://github.com/kframework/kore/blob/master/kore/src/Kore/Step/Function/Evaluator.hs#L262) so that we can directly check if an evaluator is available. It would probably be a good idea to extract that function, too.